client/components/domains/free-domain-explainer/index.jsx

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class FreeDomainExplainer extends React.Component {
+	handleClick = () => {
+		const hideFreePlan = true;
+
+		this.props.onSkip( undefined, hideFreePlan );
+	};
+	render() {
+		return (
+			<div className="free-domain-explainer card is-compact">
+				<header>
+					<h1 className="free-domain-explainer__title">
+						Get a free one-year domain registration with any paid plan.
+					</h1>
+					<p className="free-domain-explainer__subtitle">
+						We'll pay the registration fees for your new domain when you choose a paid plan during
+						the next step.
+					</p>
+					<p className="free-domain-explainer__subtitle">
+						You can claim your free custom domain later if you aren't ready yet.
+						<Button
+							borderless
+							className="free-domain-explainer__subtitle-link"
+							onClick={ this.handleClick }
+							href
```

Comment:
FYI, this line is causing a warning in the console:

```
backend.js:6 Warning: Failed prop type: Invalid prop `href` of type `boolean` supplied to `Button`, expected `string`.
```
That's because it's getting a `true` if you use that form. You should either explicitly assign an empty string value to it (`href={ '' }`) or better yet, just omit the attribute.

Reply:
Added a quickfix for it here: https://github.com/Automattic/wp-calypso/pull/39092/commits/04ab81642a17ddd3d55b51e57a07b4397f68caf4 (part of https://github.com/Automattic/wp-calypso/pull/39092) 🙇